xonsh/ptk_shell/shell.py

```diff
@@ -215,6 +216,8 @@ def prompt_tokens(self):
             p = self.prompt_formatter(p)
         except Exception:  # pylint: disable=broad-except
             print_exception()
+        if "\x1b" in p:
+            return ANSI(p)
```

Comment:
Should this check happen on each individual token? I think if xonsh coloring and ANSI coloring is mixed, it won't display correctly.

Reply:
Yep, currently it's either ANSI or *xonsh* coloring, but you're right, it might be possible to mix them if this is done after *xonsh* color tokenization, I'll give it a try.

Reply:
Awesome, thanks!

Reply:
Ok, it was a bit more complicated than I imagined, but got that finally, mixed formatting works, i.e.:
```
$PROMPT=lambda: $(starship prompt) + "\n{RED}{user}{GREEN}> "
```